docs/src/architecture/notes.md

```diff
@@ -1,82 +1,8 @@
-
-## Note hash
-The note hash is computed as:
-
-`hash(hash(hash(hash(serial_num, [0; 4]), script_hash), input_hash), vault_hash)`
-
-This achieves the following properties:
-- Every note can be reduced to a single unique hash.
-- To compute a note's hash, we do not need to know the note's `serial_num`. Knowing the hash
-    of the `serial_num` (as well as `script_hash`, `input_hash` and `note_vault`) is sufficient.
-- Moreover, we define `recipient` as: `hash(hash(hash(serial_num, [0; 4]), script_hash), input_hash)`. This allows computing the note hash from recipient and note vault.
-- We compute the hash of `serial_num` as `hash(serial_num, [0; 4])` to simplify processing within
-the VM.
-
-## Note nullifier
-The nullifier is the note's index in the [Nullifier DB](https://0xpolygonmiden.github.io/miden-base/architecture/state.html#nullifier-database). The Nullifier DB stores the information whether a note was already consumed.
-
-The nullifier is computed as `hash(serial_num, script_hash, input_hash, vault_hash)`.
-
-This achieves the following properties:
-- Every note can be reduced to a single unique nullifier.
-- We cannot derive a note's hash from its nullifier.
-- To compute the nullifier, we must know all components of the note: `serial_num`, `script_hash`, `input_hash`, and `vault_hash`.
-
-To know a note’s nullifier, one needs to know all details of the note. That means if a note is private and the operator stores only the note's hash, only those with the note details know if this note has been consumed already. Zcash first introduced this approach.
-
-<p align="center">
-    <img src="../diagrams/architecture/note/Nullifier.png">
-</p>
-
-## Note lifecycle
-For a note to exist it must be present in the [Notes DB](https://0xpolygonmiden.github.io/miden-base/architecture/state.html#notes-database) kept by the Miden Node(s). New notes are being produced when executing a transaction. They can be produced locally by users in local transactions or by the operator in a network transaction.
-
-The lifcycle of a note is as follows:
-* A new note is produced when a transaction is executed - regardless of the transaction type
-* Operator will receive the note hash and if the note is public, it'll also receive the corresponding note's data
-* Operator verifies the correctness of the underlying transaction before adding the note hash to the Notes DB
-* The note can now be consumed in a seperate transaction - to consume the note, the note's data must be known
-* A note is consumed when the its nullifier in the [Nullifier DB](https://0xpolygonmiden.github.io/miden-base/architecture/state.html#nullifier-database) is set to `1`
-* Operator will receive the note's nullifier together with a transaction proof
-* After successful verification, the Operator sets the corresponding entry in the Nullifier DB to `1`
+- [Note Design](./notes/note_design.md)
+- [Note Concepts](./notes/note_concepts.md) 
+- [Note Lifecycle](./notes/note_lifecycle.md)
```

Comment:
I'm not a huge fan of splitting the content into lots of smaller articles. We can do this of course, but I think we'd want the content of sub-chapters to be relatively stand-alone. In this specific case, I'd probably keep the core content in the main article. A good potential sub-section could be an example note (e.g., `P2ID`) or maybe more targeted concepts like note discovery, authentication etc.